File: project_euler/00/007.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const n = 200000
const k = 10001

func primeSieve(n int) []int {
	prime := []int{}
	isPrime := make([]bool, n)
	for i := 0; i < n; i++ {
		isPrime[i] = true
	}
	for i := 2; i < n; i++ {
		if isPrime[i] {
			prime = append(prime, i)
		}
		for j := 0; j < len(prime) && prime[j]*i < n; j++ {
			isPrime[prime[j]*i] = false
			if i%prime[j] == 0 {
				break
			}
		}
	}

	return prime
}

func main() {
	prime := primeSieve(n)

	kthPrime := prime[k-1]

	fmt.Printf("%d\n", kthPrime)
}

File: project_euler/00/007.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

N = 200000
K = 10001


def prime_sieve(num_n):
    prime = []
    is_prime = [True for i in range(num_n)]
    for i in range(2, num_n):
        if is_prime[i]:
            prime.append(i)
        for num_p in prime:
            if num_p * i >= num_n:
                break
            is_prime[num_p * i] = False
            if i % num_p == 0:
                break

    return prime


def main():
    prime = prime_sieve(N)

    kth_prime = prime[K - 1]

    print(kth_prime)


if __name__ == '__main__':
    main()
